handler/sigfunc.h

```diff
@@ -0,0 +1,50 @@
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef void Sigfunc(int);
+
+Sigfunc *signal(int signo, Sigfunc *func)
+{
+	struct sigaction act,oact;
+
+	act.sa_handler = func;
+	sigemptyset(&act.sa_mask);
+	act.sa_flags = 0;
+	if(signo == SIGALRM){
+#ifdef SA_INTERRUPT
```

Comment:
SA_RESTART is optional flag. When the flag is set, a system call interrupted by this signal(example accept,read,write) will be automatically restarted by the kernel.
If the signal being caught is not SIGALRM, we specify the SA_RESTART flag, if defined.The reason for making a special case for SIGALRM is that the purpose of genearting this signal is normally to place a timeout on an I/O operation, in which case, we want the blocked system call to be interrupted by the signal.